data/projects.js

```diff
@@ -230,4 +230,45 @@ implemented.",
 		"initiatives" : ["GSoC"],
 		"tags" : ["Core", "Bears", "Debugging", "pdb"]
 	}
+	{
+		"name" : "Handle languages inside other languages",
+		"desc" : "Multiple programming languages can coexist in a single \
+source file and coala would benefit from having an API to support such \
+analysis algorithms. If we have a php file, it is inevitable that we'd have \
+chunks of html inside a few of them. We want some method of finding specific \
+languages inside other languages and handle them accordingly. Like awk inside \
+bash, JSON inside Javascript, etc.",
+		"requirements" : [
+			"At least one patch to the coala core should be accepted and merged.",
+			"The student should be familiar with ongoing technical cEPs.",
+			"The proposal needs to contain a clear concept.",
+		],
+		"difficulty" : "medium",
+		"issues" : [
+			"https://github.com/coala/coala/issues/1690"
+		],
+		"mentors" : [
+			"sils",
+			"Makman2",
+			"Udayan12167"
+		],
+		"milestones" : {
+			"GSoC 2017 Coding Starts" : [
+				"The applicant has made a cEP of how multiple languages will \
+be handled inside a single source file.",
+				"The applicant is familiar with core coala codebase."
+			],
+			"GSoC 2017 Midterm" : [
+				"The API code should be ready and in testing.",
+				"An example bear using the API should be showcased.",
```

Comment:
not necessarily functional but I'd have to be able to write a bear and a prototype for the functional thing should exist IMO